synfig-studio/src/gui/widgets/widget_timeslider.cpp

```diff
@@ -338,9 +365,20 @@ Widget_Timeslider::on_button_press_event(GdkEventButton *event) //for clicking
 
 	if (event->button == 1) {
 		Time time = time_plot_data->get_t_from_pixel_coord(event->x);
-		time_plot_data->time_model->set_time(time);
+		time_plot_data->time_model->set_time(time);//setting the time which is in time model
 	}
 
+	double current= time_plot_data->get_double_pixel_t_coord(time_plot_data->get_t_from_pixel_coord(event->x));
+	double x0,x1,w,z0,z1,width;
+	get_bounds_rectangle_dimensions(x0,x1,w,true);
+	get_bounds_rectangle_dimensions(z0,z1,width,false);
+
+	if((current <= (x0+w))  && (current >= (x0 + w - boundary_dimension) ))
+		move_lower_bound_button = true;
+	else if((current <= (z0 + boundary_dimension))  && (current >= (z0)))
```

Comment:
While it's better than if they were missing, there are many parenthesis here XD
In C++ at least, only the outmost parenthesis here are needed.

Reply:
I admit they are too much here :') will remove them.